### ISSUE TITLE
ResNet101 code

### ISSUE THREAD

**Issue by GuoleiSun** [PengtaoJiang/OAA-PyTorch#2]
Hi Peng-Tao,
In your code, only vgg16 is provided. Could you also provide resnet101?
Thanks

**Comment by PengtaoJiang** [comment]
Hello, the accumulation process for vgg16 is the same with resnet101, you just need to replace the vggnet with the resnet. I will consider uploading the code later.

**Comment by SII-Ferenas** [comment]
> Hello, the accumulation process for vgg16 is the same with resnet101, you just need to replace the vggnet with the resnet. I will consider uploading the code later.

Thanks for your great work ! Have you tested your model in ResNet50 or other backbones and  what performance do they achieve ?

**Comment by PengtaoJiang** [comment]
I have not tried this.  I could test the accumulation process for resnet50.  When I obtain the performance, I will tell you. Thanks for following our work.

**Comment by SII-Ferenas** [comment]
> I have not tried this. I could test the accumulation process for resnet50. When I obtain the performance, I will tell you. Thanks for following our work.

Hi, thanks for your reply! I was wondering how you test your CAMs made by OAA, here you did not provide the test file for the mIoU of your CAM (58.6% in your paper). How do you evaluate the background score? Do you follow the common sense that treating the bg score as 1-max(foreground_score)?

**Comment by PengtaoJiang** [comment]
58.6% is not the mIoU of CAM.  It is the mIoU of the segmentation network trained using the pseudo labels generated by CAM.  In the gen_gt.py file, we utilize the 1-saliency as the background cues, not 1-max(foreground_score).

**Comment by SII-Ferenas** [comment]
> 58.6% is not the mIoU of CAM. It is the mIoU of the segmentation network trained using the pseudo labels generated by CAM. In the gen_gt.py file, we utilize the 1-saliency as the background cues, not 1-max(foreground_score).

Thanks for your reply! However, I was more confused about it. In Tab. 1, the mIoU of segmentation masks reached 61.6% in val set, but here 58.6% (data in Tab. 2) is also the segmentation masks. Could you provide more details about it ? In WSSS, the CAM or refined CAM is firstly generated, then is treated as the pseudo masks as the semantic segmentation. Therefore, I was confused about your claim.

**Comment by PengtaoJiang** [comment]
Hi, I have written that the ablation experiments are all conducted on a single test in the paper. The final segmentation results are evaluated on a multi-scale test with CRF (58.6 -> 61.6).   Moreover, the CAM and refined CAM are first used to generate pseudo segmentation labels, then the segmentation labels are used to train deeplab.

**Comment by SII-Ferenas** [comment]
I got that！Thanks for your reply！

发自我的iPhone
